### ISSUE TITLE
Rules that verify correct initialization of objects should not raise twice

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#3129]
The rules that are raising based on correct initialization of objects , based on `ObjectShouldBeInitializedCorrectlyBase`, will raise two issues for the same code problem when both the constructor is unsafe and an unsafe property is set.

Repro steps for `CookieShouldBeHttpOnly`

```
        void RaiseTwice()
        {
            var x = new HttpCookie("c"); // Noncompliant {{Make sure creating this cookie without the "HttpOnly" flag is safe.}}
            x.HttpOnly = false; // Noncompliant
        }
```

Expected: should raise only once.

**Comment by pavel-mikula-sonarsource** [comment]
The problem could be on two places further away from each other, or maybe with some if/else/nested scenario. Making it hard to track, so reporting twice is probably the safest way of doing this.

If somebody does the same mistake twice, they'd at least get a hint that the 2nd command one is pointless.
